snuba/datasets/dataset_tables.py

```diff
@@ -0,0 +1,73 @@
+class DatasetTables(object):
+    """
+    A collection of tables associated with a dataset, used to obfuscate methods from the schema
+    """
+
+    def __init__(self, read_schema, write_schema, intermediary_schemas=[]):
+        self.__read_schema = read_schema
+        self.__write_schema = write_schema
+        self.__intermediary_schemas = intermediary_schemas
+
+    def get_local_write_table_name(self):
+        return self.__write_schema.get_local_table_name()
+
+    def get_local_read_table_name(self):
+        return self.__read_schema.get_local_table_name()
+
+    def get_write_table_name(self):
+        return self.__write_schema.get_table_name()
+
+    def get_read_table_name(self):
+        return self.__read_schema.get_table_name()
+
+    def get_read_columns(self):
+        return self.__read_schema.get_columns()
+
+    def get_write_columns(self):
+        return self.__write_schema.get_columns()
+
+    def __get_unique_schemas(self):
+        all_schemas_with_possible_duplicates = [self.__read_schema, self.__write_schema] + self.__intermediary_schemas
+
+        return list(set(all_schemas_with_possible_duplicates))
+
+    def get_all_table_names(self):
+        # TODO: remove get_all_table_names method once we not longer need it in the health check
+        return map(lambda schema: schema.get_table_name(), self.__get_unique_schemas())
+
+    def get_all_local_table_names(self):
```

Comment:
This is only used for migrations. I've kept it here so that `migrate.py` can continue to be used. But depending on how we do migrations in the future, we could remove this method.